src/transfer.ts

```diff
@@ -74,3 +74,53 @@ export async function ensureFunded(funding: EnsureFundedParams, algod: Algodv2,
 
   return undefined
 }
+
+interface TransferAssetParams extends SendTransactionParams {
+  senderAccount: SendTransactionFrom
+  receiverAccount: Account
+  assetId: number
+  amount: number | bigint
+  transactionParams?: SuggestedParams
+  clawbackFrom?: SendTransactionFrom
+  note?: TransactionNote
+}
+
+export async function transferAsset(
+  { senderAccount, receiverAccount, assetId, amount, transactionParams, clawbackFrom, note, ...sendParams }: TransferAssetParams,
+  algod: Algodv2,
+) {
+  const transaction = algosdk.makeAssetTransferTxnWithSuggestedParamsFromObject({
+    from: getSenderAddress(senderAccount),
+    to: receiverAccount.addr,
+    closeRemainderTo: undefined,
+    revocationTarget: clawbackFrom ? getSenderAddress(clawbackFrom) : undefined,
+    amount: amount,
+    note: encodeTransactionNote(note),
+    assetIndex: assetId,
+    suggestedParams: await getTransactionParams(transactionParams, algod),
+    rekeyTo: undefined,
+  })
+
+  if (!sendParams.skipSending) {
+    if (sendParams.maxFee === undefined) {
+      sendParams.maxFee = (0.02).algos()
+    }
+
+    const result = await sendTransaction(
+      {
+        transaction: transaction,
+        from: senderAccount,
+        sendParams: sendParams,
+      },
+      algod,
+    )
+
+    Config.getLogger(sendParams.suppressLog).info(
+      `Successfully transferred ${amount} of asset ${assetId} from ${getSenderAddress(senderAccount)} to ${receiverAccount}.`,
+    )
+
+    return result
+  }
+
+  return { transaction, confirmation: undefined }
```

Comment:
I think reusing the conditional and return statement similar to how it's implemented in transferAlgos will be a better option here.